autometa/common/kmers.py

```diff
@@ -41,12 +41,27 @@
 from umap import UMAP
 
 from autometa.common.utilities import gunzip
+from autometa.common.exceptions import KmerFormatError
+from autometa.common.exceptions import KmerEmbeddingError
 
-logger = logging.getLogger(__name__)
 # Suppress numba logger debug output
 numba_logger = logging.getLogger("numba").setLevel(logging.ERROR)
+logger = logging.getLogger(__name__)
 
 def revcomp(string):
+    """Revers complement the provided `string`.
+
+    Parameters
+    ----------
+    string : str
+        Description of parameter `string`.
+
+    Returns
+    -------
+    str or int(-1)
+        reverse complemented string.
+        Note: If any weird letters are encountered, int value of -1 is returned.
```

Comment:
Nitpick, but in the comment, I would include that the reason for this is so that we don't count k-mers with weird characters. Otherwise someone in the future might decide that this needs "fixing".

Reply:
This exception will _never_ be encountered. `revcomp()` is only used once and this is used on the `all_kmers` variable which is initialized in the script. Therefore, perhaps for more clarity, this could be removed entirely.

Reply:
Maybe. I just thought it would be worth including a comment about our reasoning. Otherwise it is a bit difficult to grasp from code review alone.

Reply:
I've simplified the function to make it intuitive to the reader. 

I.e. complement the character for every character in the string in reverse order

I've also removed the unnecessary exception handling and denoted this a function used only as an internal by prepending the underscore (conventions to the programming community).